presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1166,28 +1165,56 @@ protected Type visitParameter(Parameter node, StackableAstVisitorContext<Context
         protected Type visitExtract(Extract node, StackableAstVisitorContext<Context> context)
         {
             Type type = process(node.getExpression(), context);
-            if (!isDateTimeType(type)) {
-                throw semanticException(TYPE_MISMATCH, node.getExpression(), "Type of argument to extract must be DATE, TIME, TIMESTAMP, or INTERVAL (actual %s)", type);
-            }
             Extract.Field field = node.getField();
-            if ((field == TIMEZONE_HOUR || field == TIMEZONE_MINUTE) && !(type instanceof TimeWithTimeZoneType || type instanceof TimestampWithTimeZoneType)) {
-                throw semanticException(TYPE_MISMATCH, node.getExpression(), "Type of argument to extract time zone field must have a time zone (actual %s)", type);
+
+            switch (field) {
```

Comment:
add `default` branch with throw

however, i think it would be better to impl this logic in the Extrac.Field enum.
the enum could take `Set<Class<Type>> supportedTypes`

Reply:
> however, i think it would be better to impl this logic in the Extrac.Field enum.
the enum could take Set<Class<Type>> supportedTypes

That would make the parser dependent on types. It's structurally not possible (there would be a circular dependency between modules), but also not appropriate, as it's something the AST/Parser is not concerned about.